src/main/scala/com/microsoft/hyperspace/index/IndexLogEntry.scala

```diff
@@ -45,31 +46,20 @@ case class Content(root: Directory, fingerprint: NoOpFingerprint = NoOpFingerpri
   @JsonIgnore
   lazy val files: Seq[Path] = {
     // Recursively find files from directory tree.
-    rec(new Path(root.name), root, (f, prefix) => new Path(prefix, f.name))
+    recFilesApply(new Path(root.name), root, (f, prefix) => new Path(prefix, f.name))
   }
 
   @JsonIgnore
   lazy val fileInfos: Set[FileInfo] = {
-    rec(
-      new Path(root.name),
-      root,
-      (f, prefix) =>
-        FileInfo(new Path(prefix, f.name).toString, f.size, f.modifiedTime, f.id)).toSet
-  }
-
-  private def rec[T](
-      prefixPath: Path,
-      directory: Directory,
-      func: (FileInfo, Path) => T): Seq[T] = {
-    val files = directory.files.map(f => func(f, prefixPath))
-    files ++ directory.subDirs.flatMap { dir =>
-      rec(new Path(prefixPath, dir.name), dir, func)
-    }
+    recFilesApply(new Path(root.name), root, createFileInfoWithPrefix).toSet
   }
 }
 
 object Content {
 
+  val createFileInfoWithPrefix: (FileInfo, Path) => FileInfo = (f, prefix) =>
```

Comment:
just impl. in `fileInfos` - there's no other use case.

Reply:
Correct. I`m just not keen of formatted:
```
      (
          f,
          prefix) =>
        FileInfo(new Path(prefix, f.name).toString, f.size, f.modifiedTime, f.id)).toSet
```
see: https://github.com/microsoft/hyperspace/pull/412/files#diff-5c50f07a41423f4cb5bfe60ac7c30b9c534c4bfc71eeb03d43040047a3994e83

Moving it to separate function makes it more readable.